src/OpenTelemetry/BatchExportProcessor.cs

```diff
@@ -198,7 +198,7 @@ protected override bool OnForceFlush(int timeoutMilliseconds)
                     return true;
                 }
 
-                if (this.shutdownDrainTarget != long.MaxValue)
+                if (Volatile.Read(ref this.shutdownDrainTarget) != long.MaxValue)
```

Comment:
Wondering why do we use `Volatile.Read`/`Volatile.Write` to get atomic semantic. As per the doc below, both functions are for consumed by compiler to decide inserting memory barrier or not, and the `InterLocked` instruction could have the side-effect of inserting barrier. I also looked at the implementation in the runtime shared here, which says 32-bit ordinary read is not atomic so lock instruction required. Not sure if it implies that ordinary read of 32-bit is atomic on x64, or even on ARM64, the behavior may be very different and could still required explicit lock.

https://learn.microsoft.com/en-us/dotnet/api/system.threading.volatile.read?view=net-8.0